modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue

```diff
@@ -177,7 +185,12 @@ export default {
     getDefault() {
       const widget = {};
       this.schema.forEach(field => {
-        widget[field.name] = field.def ? klona(field.def) : field.def;
+        // Do not simply check if `field.def` is truthy, this in not enough
+        // since a def as an empty string must be considered valid:
+        const hasDefaultValue = Object.prototype.hasOwnProperty.call(field, 'def');
+        widget[field.name] = hasDefaultValue
+          ? klona(field.def)
+          : null;
```

Comment:
this is a common annoying bug: we were simply checking if `field.def` was truthy, therefore `false`, `null`, `''`, `0` and every falsy values were not taken into account

Reply:
💯